### PR TITLE
JS: explain camel case & ES6 rules

### PR DESCRIPTION
**Pull request by wfleming**
@codeclimate/review this copies over what was added & discussed on `browser`.

### CODE REVIEW

File: javascript/eslintrc-es6.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  "rules": {
+    "consistent-this": [2, "prefer-fat-arrow-over-reassigning-thing"],
```
Comment (pbrisbin) [inline javascript/eslintrc-es6.js]: Did you mean "...reassigning-thing", or "...reassigning-this"?

Comment (wfleming) [inline javascript/eslintrc-es6.js]: `this`. Already fixed.

**Comment by pbrisbin** [comment]
This all LGTM. I think we have a loose requirement of a few LGTMs on styleguide stuff tough.

File: javascript/eslintrc-es6.js (hunk 1 above)
Comment (wfleming) [inline javascript/eslintrc-es6.js]: I  mean, you should't reassign thing either, probably.

**Comment by GordonDiggs** [comment]
I'm +1 on all of this too

**Comment by wfleming** [comment]
This was all basically discussed before hand anyway: I'm calling 2 +1s good enough.

**Pull request merged**